### PR TITLE
More teleport timing adjustments

### PR DESCRIPTION
**Pull request by TeoTwawki**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
This is a follow-up on #4770 with more corrections. Its in draft till I verify a few things.

## Steps to test these changes
try rings of recall and teleport, see that the full animations plays before you get zoned.

### CODE REVIEW

**Review by Xaver-DaRed — COMMENTED** [review]
I like what I'm seing, even if ts a draft. Specially the change to enhacing magic file.

**Comment by TeoTwawki** [comment]
All done and re-tested last revision.

**Review by Xaver-DaRed — APPROVED** [review]

File: scripts/items/treat_staff.lua [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+local itemObject = {}
+
+itemObject.onItemCheck = function(target)
+    return 0
+end
+
+itemObject.onItemEquip  = function(user, item)
+    user:addListener('ATTACK', 'TREAT_STAFF_MELEE', function(player, target, action)
+        if
+            math.random(1,100) <= 10 and
```
Comment (Xaver-DaRed) [inline scripts/items/treat_staff.lua]: missing space

Comment (TeoTwawki) [reply scripts/items/treat_staff.lua]: REEEEEEEEEEEEEEEEEEE

**Review by WinterSolstice8 — APPROVED** [review]

**Pull request merged**